lib/common/strings.c

```diff
@@ -602,38 +602,52 @@ crm_strdup_printf(char const *format, ...)
 }
 
 int
-pcmk__split_range(const char *srcstring, char separator, char **start, char **end)
+pcmk__split_range(const char *srcstring, long long *start, long long *end)
 {
-    const char *seploc = NULL;
+    char *remainder = NULL;
+
+    *start = -1;
+    *end = -1;
 
     CRM_ASSERT(start != NULL && end != NULL);
-    *start = NULL;
-    *end = NULL;
 
     crm_trace("Attempting to decode: [%s]", srcstring);
-    if (srcstring == NULL || strcmp(srcstring, "") == 0) {
+    if (srcstring == NULL || strcmp(srcstring, "") == 0 || strcmp(srcstring, "-") == 0) {
         return pcmk_rc_unknown_format;
     }
 
-    seploc = strchr(srcstring, separator);
-    if (!seploc) {
-        return pcmk_rc_ok;
-    } else if (strlen(srcstring) == 1) {
-        /* The source string contained only the separator. */
+    /* String starts with a dash, so this is either a range with
+     * no beginning or garbage.
+     * */
+    if (*srcstring == '-') {
+        int rc = scan_ll(srcstring+1, end, &remainder);
+
+        if (rc != pcmk_rc_ok || *remainder != '\0') {
+            return pcmk_rc_unknown_format;
+        } else {
+            return pcmk_rc_ok;
+        }
+    }
+
+    if (scan_ll(srcstring, start, &remainder) != pcmk_rc_ok) {
         return pcmk_rc_unknown_format;
-    } else if (seploc == srcstring && *(seploc + 1)) {
-        /* Separator is the first character of the range, so this
-         * range only has an end.
-         */
-        *end = strdup(seploc + 1);
-    } else if (! *(seploc + 1)) {
-        /* Separator is the last character of the range, so this
-         * range only has a start.
-         */
-        *start = strndup(srcstring, seploc - srcstring);
+    }
+
+    if (*remainder && *remainder == '-') {
+        if (*(remainder+1)) {
+            char *more_remainder = NULL;
+            int rc = scan_ll(remainder+1, end, &more_remainder);
+
+            if (rc != pcmk_rc_ok || *more_remainder != '\0') {
+                return pcmk_rc_unknown_format;
+            }
+        }
     } else {
-        *start = strndup(srcstring, seploc - srcstring);
-        *end = strdup(seploc + 1);
+        /* The string contained only one number.  Reset start to -1 and
+         * return pcmk_rc_ok.  This gives the caller a way to tell this
+         * condition apart from a range with no end.
+         */
+        *start = -1;
```

Comment:
Why not just set *end = *start? "1" and "1-1" are logically equivalent

Reply:
So, the caller would check for pcmk_rc_ok and start == end?  That would probably simplify things.  The caller wouldn't have to call scan_ll again, at least.  Sounds good.